Panic in DirectFailure Marshalling

```
panic: runtime error: invalid memory address or nil pointer dereference, stack: goroutine 114 [running]:
runtime/debug.Stack()
        /layers/google.go.gomod/gopath/pkg/mod/[golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/runtime/debug/stack.go:26](http://golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/runtime/debug/stack.go:26) +0x5e
[github.com/ausocean/cloud/utils.NewConfigurableRecoveryHandler.func2({0xd13780](http://github.com/ausocean/cloud/utils.NewConfigurableRecoveryHandler.func2(%7B0xd13780), 0x1618930})
        /layers/google.go.appengine-gomod/srv/utils/utils.go:175 +0x1d
[github.com/ausocean/cloud/utils.NewConfigurableRecoveryHandler.func4({0xfc26c0](http://github.com/ausocean/cloud/utils.NewConfigurableRecoveryHandler.func4(%7B0xfc26c0), 0xc0003f2b60}, {0xd13780, 0x1618930})
        /layers/google.go.appengine-gomod/srv/utils/utils.go:190 +0x76
[github.com/ausocean/cloud/utils.(*RecoverableServeMux).ServeHTTP.func1()](http://github.com/ausocean/cloud/utils.(*RecoverableServeMux).ServeHTTP.func1())
        /layers/google.go.appengine-gomod/srv/utils/utils.go:242 +0x62
panic({0xd13780?, 0x1618930?})
        /layers/google.go.gomod/gopath/pkg/mod/[golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/runtime/panic.go:785](http://golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/runtime/panic.go:785) +0x132
encoding/json.(*encodeState).marshal.func1()
        /layers/google.go.gomod/gopath/pkg/mod/[golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/encoding/json/encode.go:294](http://golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/encoding/json/encode.go:294) +0x6d
panic({0xd13780?, 0x1618930?})
        /layers/google.go.gomod/gopath/pkg/mod/[golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/runtime/panic.go:785](http://golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/runtime/panic.go:785) +0x132
main.(*directFailure).MarshalJSON(0x161b030?)
        /layers/google.go.appengine-gomod/srv/cmd/oceantv/state_direct_failure.go:63 +0x15
encoding/json.marshalerEncoder(0xc0004653c0, {0xd88620?, 0xc0004f9f50?, 0x0?}, {0x0?, 0x0?})
        /layers/google.go.gomod/gopath/pkg/mod/[golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/encoding/json/encode.go:441](http://golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/encoding/json/encode.go:441) +0xd2
encoding/json.(*encodeState).reflectValue(0xc0004653c0, {0xd88620?, 0xc0004f9f50?, 0xc0001e78f0?}, {0xe0?, 0xf1?})
        /layers/google.go.gomod/gopath/pkg/mod/[golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/encoding/json/encode.go:322](http://golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/encoding/json/encode.go:322) +0x73
encoding/json.(*encodeState).marshal(0x0?, {0xd88620?, 0xc0004f9f50?}, {0x68?, 0xcd?})
        /layers/google.go.gomod/gopath/pkg/mod/[golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/encoding/json/encode.go:298](http://golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/encoding/json/encode.go:298) +0xbd
encoding/json.Marshal({0xd88620, 0xc0004f9f50})
        /layers/google.go.gomod/gopath/pkg/mod/[golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/encoding/json/encode.go:164](http://golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/encoding/json/encode.go:164) +0xbe
main.updateBroadcastBasedOnState({0xfbf2e0, 0xc0004f9f50}, 0xc000601b88)
        /layers/google.go.appengine-gomod/srv/cmd/oceantv/broadcast_states.go:317 +0x53d
main.(*broadcastStateMachine).handleEvent.func1(0xc00017e480?)
        /layers/google.go.appengine-gomod/srv/cmd/oceantv/broadcast_machine.go:101 +0x25
main.(*OceanBroadcastManager).Save.func2(0xc000601b88)
        /layers/google.go.appengine-gomod/srv/cmd/oceantv/broadcast_manager.go:239 +0x29
main.updateConfigWithTransaction.func1({0xfbef48?, 0xc0005af4f0})
        /layers/google.go.appengine-gomod/srv/cmd/oceantv/utils.go:151 +0xf3
[github.com/ausocean/openfish/datastore.(*CloudStore).Update.func1(0xc000690e60)](http://github.com/ausocean/openfish/datastore.(*CloudStore).Update.func1(0xc000690e60))
        /layers/google.go.gomod/gopath/pkg/mod/[github.com/ausocean/openfish@v0.1.6/datastore/datastore.go:349](http://github.com/ausocean/openfish@v0.1.6/datastore/datastore.go:349) +0x6a
[cloud.google.com/go/datastore.(*Client).RunInTransaction(0xc000464440](http://cloud.google.com/go/datastore.(*Client).RunInTransaction(0xc000464440), {0xfc4e58?, 0x16649a0?}, 0xc00034d048, {0x0, 0x0, 0x0})
        /layers/google.go.gomod/gopath/pkg/mod/[cloud.google.com/go/datastore@v1.11.0/transaction.go:202](http://cloud.google.com/go/datastore@v1.11.0/transaction.go:202) +0x14f
[github.com/ausocean/openfish/datastore.(*CloudStore).Update(0x0](http://github.com/ausocean/openfish/datastore.(*CloudStore).Update(0x0)?, {0xfc4e58?, 0x16649a0?}, 0x412525?, 0x50?, {0xfbef48?, 0xc0005af4f0?})
        /layers/google.go.gomod/gopath/pkg/mod/[github.com/ausocean/openfish@v0.1.6/datastore/datastore.go:344](http://github.com/ausocean/openfish@v0.1.6/datastore/datastore.go:344) +0x65
main.(*CompositeStore).Update(0xc00011cfc0?, {0xfc4e58, 0x16649a0}, 0xc0005a36c0, 0xc00047b068, {0xfbef48, 0xc0005af4f0})
        /layers/google.go.appengine-gomod/srv/cmd/oceantv/store.go:144 +0x65
main.updateConfigWithTransaction({0xfc4e58, 0x16649a0}, {0xfca330, 0xc000459da0}, 0x72b957c8d1, {0xc00011c600?, 0xc078bd?}, 0xc00047b050)
        /layers/google.go.appengine-gomod/srv/cmd/oceantv/utils.go:163 +0x223
main.(*OceanBroadcastManager).Save(0xc00051ec30, {0x0, 0x0}, 0xc0005a7360)
        /layers/google.go.appengine-gomod/srv/cmd/oceantv/broadcast_manager.go:247 +0x655
main.(*broadcastStateMachine).handleEvent(0xc000123d60, {0xfb7e00?, 0xc00047b020?})
        /layers/google.go.appengine-gomod/srv/cmd/oceantv/broadcast_machine.go:101 +0x302
main.(*basicEventBus).publish(0xc0005b0480, {0xfb7e00, 0xc00047b020})
        /layers/google.go.appengine-gomod/srv/cmd/oceantv/broadcast_events.go:463 +0x16a
main.(*broadcastSystem).tick(0xc0001e7320)
        /layers/google.go.appengine-gomod/srv/cmd/oceantv/system.go:267 +0x5c2
main.performChecksInternalThroughStateMachine({0xfc4ec8?, 0xc000149770?}, 0xc0005b6008, 0x527079?, {0xfca330, 0xc000459da0}, {0xc000124450, 0x1, 0xc00033b320?}, {0xc000124458, ...})
        /layers/google.go.appengine-gomod/srv/cmd/oceantv/broadcast.go:320 +0x478
main.performChecks({0xfc4ec8?, 0xc000149770?}, 0xe82bf6?, {0xfca330?, 0xc000459da0?}, {0xc000124450?, 0xc000106180?, 0x4864bd?}, {0xc000124458, 0x1, ...})
        /layers/google.go.appengine-gomod/srv/cmd/oceantv/broadcast.go:332 +0x71
main.checkBroadcastsForSites({0xfc4ec8, 0xc000149770}, {0xc00034d940?, 0xc000149180?, 0x0?}, {0xc000124450, 0x1, 0x1}, {0xc000124458, 0x1, ...})
        /layers/google.go.appengine-gomod/srv/cmd/oceantv/broadcast.go:271 +0x5f2
main.(*oceanTVService).checkBroadcastsHandler(0xc00045f2c0, {0xfc26c0, 0xc0003f2b60}, 0xc0000f0c80)
        /layers/google.go.appengine-gomod/srv/cmd/oceantv/broadcast.go:235 +0x525
net/http.HandlerFunc.ServeHTTP(0xc0003f29a0?, {0xfc26c0?, 0xc0003f2b60?}, 0x10?)
        /layers/google.go.gomod/gopath/pkg/mod/[golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/net/http/server.go:2220](http://golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/net/http/server.go:2220) +0x29
net/http.(*ServeMux).ServeHTTP(0x44695f?, {0xfc26c0, 0xc0003f2b60}, 0xc0000f0c80)
        /layers/google.go.gomod/gopath/pkg/mod/[golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/net/http/server.go:2747](http://golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/net/http/server.go:2747) +0x1ca
[github.com/ausocean/cloud/utils.(*RecoverableServeMux).ServeHTTP(0x46ab79](http://github.com/ausocean/cloud/utils.(*RecoverableServeMux).ServeHTTP(0x46ab79)?, {0xfc26c0?, 0xc0003f2b60?}, 0xc00008ab70?)
        /layers/google.go.appengine-gomod/srv/utils/utils.go:249 +0x56
net/http.serverHandler.ServeHTTP({0xc00045fce0?}, {0xfc26c0?, 0xc0003f2b60?}, 0x6?)
        /layers/google.go.gomod/gopath/pkg/mod/[golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/net/http/server.go:3210](http://golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/net/http/server.go:3210) +0x8e
net/http.(*conn).serve(0xc00013f560, {0xfc4e90, 0xc00045f5f0})
        /layers/google.go.gomod/gopath/pkg/mod/[golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/net/http/server.go:2092](http://golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/net/http/server.go:2092) +0x5d0
created by net/http.(*Server).Serve in goroutine 1
        /layers/google.go.gomod/gopath/pkg/mod/[golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/net/http/server.go:3360](http://golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64/src/net/http/server.go:3360) +0x485
```

--- a/cmd/oceantv/main.go
+++ b/cmd/oceantv/main.go
@@ -46,7 +46,7 @@ import (
 
 const (
 	projectID            = "oceantv"
-	version              = "v0.13.0"
+	version              = "v0.13.1"
 	projectURL           = "https://tv.cloudblue.org"
 	cronServiceAccount   = "oceancron@appspot.gserviceaccount.com"
 	locationID           = "Australia/Adelaide" // TODO: Use site location.

--- a/cmd/oceantv/state_direct_failure.go
+++ b/cmd/oceantv/state_direct_failure.go
@@ -60,7 +60,11 @@ func (s *directFailure) enter() {
 }
 
 func (s *directFailure) MarshalJSON() ([]byte, error) {
-	return json.Marshal(struct{ Err string }{Err: s.err.Error()})
+	if s.err != nil {
+		return json.Marshal(struct{ Err string }{Err: s.err.Error()})
+	}
+
+	return json.Marshal(struct{ Err string }{Err: ""})
 }
 
 func (s *directFailure) UnmarshalJSON(data []byte) error {

